explainer.md

```diff
@@ -89,21 +89,29 @@ The second mode where styles are copied to a container element won't be part of
+### Retaining hierarchy during a transition
+
+By default, offered elements are captured in a flat hierarchy. As in all offered elements, including the root, will be laid out as siblings. The browser uses cached transforms to position each element so it overlaps exactly with its quad on the old page.
+
+The means that, during the transition, scaling one offered element won't impact the rendering of another offered element, even if it was a child of the other element in the old page.
+
+This also allows elements to visually move between containers in viewport space, even if they were clipped to some parent in old page.
+
+Alternatively, the developer can make an offered element a 'transition container'. Offered elements will be nested within their closest transition container, and the cached transform will position the element within that container rather than the viewport. This is similar in spirit to how `position: relative` creates a new 0,0 for absolutely positioned elements.
+
+'Transition containers' are out of scope for 'v1' of the feature. See [issue](https://github.com/WICG/shared-element-transitions/issues/74) for detailed discussion.
 
 ## Part 2: The preparation
 
 At this point the state has changed over to Page-B, and Page-A is gone aside from the elements it offered. In the MPA case, this happens when the navigation is complete. In the SPA case, this happens when the DOM is in the Page-B state and the developer signals that the change is complete (how to make that signal is discussed later in the API).
 
 ### Setting the stage
 
-The offered elements from Page-A are fixed position at (0,0) in the [top layer](https://fullscreen.spec.whatwg.org/#top-layer), and moved into their previous viewport-relative positions using the cached transform.
-
-The top layer content ensures that the user continues to see Page-A's visuals as Page-B is loading. Note that this may not reproduce the exact rendering on Page-A. For example, the relative paint order of shared elements is preserved in the top layer. But if a shared element was occluded by another element, the latter is painted into the root's image unless it is also offered as a shared element.
+The stage is a viewport-filling container with fixed position. The offered elements from Page-A are positioned absolutely at (0,0), nested according in their closest 'transition container' or the stage if there's no parent container, and moved into their previous viewport-relative positions using the cached transform. Their content is painted on top of Page-B which ensures that the user continues to see Page-A's visuals as Page-B is loading. Note that this may not reproduce the exact rendering on Page-A. For example, the relative paint order of shared elements is preserved when rendered on top of Page-B. But if a shared element was occluded by another element, the latter is painted into the root's image unless it is also offered as a shared element.
```

Comment:
@khushalsagar in my last PR I had the offered elements absolutely positioned in a fixed position container, and we switched it back to everything being fixed position. However, I think we need absolute positioning else the nesting doesn't make sense. We want transition elements to be positioned relative to 0,0 of their transition container, which may be the viewport, but may not be. Making them fixed position would put everything at 0,0 in viewport space, which doesn't seem to work.

Reply:
Regarding fixed positioning; the containing block for a fixed position element is usually the viewport, but not always (eg, setting `transform:` on an element establishes a containing block for all positioned descendants, including fixed [1]). If we want to describe elements as being fixed with respect to its nearest shared ancestor, I think we could use the same type of prose.

[1] https://www.w3.org/TR/css-transforms-1/#:~:text=the%20transform%20property%20also%20causes%20the%20element%20to%20establish%20a%20containing%20block%20for%20all%20descendants.

Reply:
Using a fixed position root for everything SGTM. We actually also needed that to describe the paint order of transition elements (using z-index). The transition DOM I'm expecting looks something like this :

transition root (position: fixed)
|_ container (position: absolute)
    |_ wrapper
        |_ old image
        |_ new image

Where the transition root has a flat list of containers for each shared element + the root image. If one of the shared elements is offered as the "transition container" for nesting then you'd have something like this :

transition root (position: fixed)
|_ container (position: absolute)
    |_ wrapper
    |   |_ old image
    |   |_ new image
    |_ container (position: absolute)
        |_ wrapper
            |_ old image
            |_ new image

So the container has nested container elements inside it. Does that sound right to you?

Reply:
Say we have a transition element that's position:fixed -- will its containing block still be the viewport? If that's true, it seems like we could be in a situation where a transition element's transform/positional parent is different than its effect parent. I.e., we'd need something like property trees to represent the relationships and things become far more complex. Could we instead make it be the case that all non-leaf nodes in this transition hierarchy establish a containing block for all positioned descendants?

Also, since this seems a bit involved, could we perhaps add a TODO for clarifying the details of the hierarchical mode and discuss this on a separate, dedicated PR?

Reply:
"we could be in a situation where a transition element's transform/positional parent is different than its effect parent". Do you mean if the developer overrides anything in this DOM to be position:fixed? If we're building this using DOM elements (or pseudo-elements) then why would it be difficult if the positional parent is different than the effect or clip parent. Are you concerned about the "out of process" implementation getting complicated?

Reply:
> Do you mean if the developer overrides anything in this DOM to be position:fixed?

Yep, that's right.

> If we're building this using DOM elements (or pseudo-elements) then why would it be difficult if the positional parent is different than the effect or clip parent. Are you concerned about the "out of process" implementation getting complicated?

I am, yeah. If engines want to support this outside of renderer processes, than this complexity would need to be replicated.

Reply:
The default hierarchy and styles set up by the UA (as described above) should ensure an element's transform parent is the same as its effect/clip parent. This only becomes an issue if the developer can modify the style on these elements and that's allowed for the renderer process implementation (same-origin) only. So it won't impact the out of process implementation.

Reply:
Could this not be impacted by the styles on transition elements even in the implicit set up for cross-origin? Or are you saying that we'd override the positioning to be `position:absolute` in that case and the only way it would change is if the developer explicitly changed that via JS on the destination page?

Reply:
Yes. We're setting the position for all container elements to be position: absolute with a UA stylesheet. Developers can change it only for same-origin via CSS or script.

Reply:
Got it. Thanks!

Reply:
@khushalsagar lol yeah the DOM model I had doesn't make sense. Not sure what I was thinking.

Reply:
I reworked the DOM stuff (which meant mostly reverting it back to what it was.